detectors/host/host_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package host
+
+import (
+	"context"
+	"os"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+)
+
+func Test_Detect(t *testing.T) {
+	detector := NewResourceDetector()
+
+	hostResource, err := detector.Detect(context.Background())
+
+	assert.True(t, err == nil)
+
+	hostName, _ := os.Hostname()
```

Comment:
This error should not be ignored. The test should be failed.

Reply:
I adapted the tests so that the `host.name` is only added to the list of expected attributes when `os.Hostname()` doesn't return an error.

This covers cases where `os.Hostname()` returns an error, but a `host.id` might still be obtained.

I'm happy to both make the tests and detector logic fail when `os.Hostname()` fails, if folks think that's better.